atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1940,7 +1948,8 @@ protected void throwIfReadWriteConflictForSerializable(long commitTimestamp) {
     }
 
     private boolean hasWrites() {
-        return writesByTable.values().stream().anyMatch(writesForTable -> !writesForTable.isEmpty());
+        return !writesByTable.isEmpty()
```

Comment:
This should avoid some of the expense of creating values & stream for read only transactions